data_safe_haven/infrastructure/stacks/shm/firewall.py

```diff
@@ -375,8 +379,30 @@ def __init__(
                 ),
                 network.AzureFirewallApplicationRuleCollectionArgs(
                     action=network.AzureFirewallRCActionArgs(type="Allow"),
-                    name=f"{stack_name}-sre-package-repositories",
+                    name=f"{stack_name}-sre-identity-servers",
                     priority=1100,
+                    rules=[
+                        network.AzureFirewallApplicationRuleArgs(
+                            description="Allow external OAuth login requests",
+                            name="AllowExternalOAuthLogin",
+                            protocols=[
+                                network.AzureFirewallApplicationRuleProtocolArgs(
+                                    port=443,
+                                    protocol_type="Https",
+                                )
+                            ],
+                            source_addresses=sre_identity_server_subnets,
+                            target_fqdns=[
+                                "graph.microsoft.com",
+                                "login.microsoftonline.com",
+                            ],
+                        ),
+                    ],
+                ),
+                network.AzureFirewallApplicationRuleCollectionArgs(
+                    action=network.AzureFirewallRCActionArgs(type="Allow"),
+                    name=f"{stack_name}-sre-package-repositories",
+                    priority=1110,
```

Comment:
Are these priorities not in an `Enum` like the NSG rules?

Reply:
Done